aptos-move/framework/supra-framework/sources/poel/poel.move

```diff
@@ -0,0 +1,308 @@
+     struct LiquidityTableItems {
+        ///Asset_name
+        asset_name: vector<u8>,
+        ///asset_supply: Total amount of the asset that has been bridged to create the iAsset.
+        asset_supply: u64,
+        ///iAsset_supply: Total supply of iAssets generated through the scheme.
+        iAsset_supply: u64,
+        ///collaterisation_rate: Specific rate indicating the collateral security of the iAsset.
+        collateralisation_rate: u64,
+        ///desired_weights: Weights reflecting the asset's strategic importance.
+        desired_weight: u64,
+        ///desirability_score: Attractiveness score of the asset.
+        desirability_score: u64,
+        ///asset_price: Current market price of the asset.
+        asset_price: u64,
+    }
+ 
+    /// This struct aggregates comprehensive liquidity metrics and operational indices for assets managed within the system.
+    struct TotalLiquidity {
+        //TotalLiquidityTable: Maps each AssetID to a LiquidityTableItems struct
+        TotalLiquidityTable: SmartTable<u64, LiquidityTableItems>,
+        //Index of the last observable lockup cycle (OLC), crucial for the minting of preminted tokens and the withdrawal of unlocked tokens.
+        current_olc_index: u64,
+        //Represents the nominal value of all assets submitted to the system, indicating the total economic stake.
+        total_nominal_value: u64
+    }
+ 
+    /// This struct is applied for the dynamic adjustment of the system's operational parameters. 
+    /// It allows for flexible management of financial metrics, adapting to market conditions or strategic shifts in policy.
+    struct MutableParameters { 
+        coefficient_b: u64,
+        coefficient_X: u64,
+        coefficient_rho: u64, 
+        min_collateralisation: u64 
+    }
+
+
+    struct StakingPoolMap {
+        //Address of the staking pool.
+        pool_address: address,
+        //The amount of Supra delegated to the validator at this pool.
+        delagated_amount: u64,
+        //Amount of tokens that are pending deactivation but are currently inactive.
+        pending_inactive_balance: u64
+    }
+
+    /// Description: This struct serves as a central repository for tracking delegated assets in various staking pools, 
+    /// providing essential data for managing staking operations and calculating borrowing limits based on active and 
+    /// inactive balances within the system.
+    struct DelegatedAmount {
+        // Maps StakingPoolID to tuple StakingPoolMap
+        staking_pool_mapping: table::Table<u64, StakingPoolMap>,
+        // Cumulative amount of Supra delegated across all staking pools
+        total_delegated_amount: u64,
+        // Total amount of Supra pending deactivation    
+        pending_inactive_balance: u64,
+        // Total amount of Supra that could be borrowed from the PoEL contract 
+        total_borrowable_amount: u64,
```

Comment:
out of? do we have a parameter that defines the max?

Reply:
Total amount will be based on the supra reserve in the contract which acts as the cap.